src/components/Card.js

```diff
@@ -9,9 +12,17 @@
       this._cardElement.querySelector('.card__link').addEventListener('click', () => {
         this._handleLikeButton();
       });
+      
       this._cardElement.querySelector('.card__delete-btn').addEventListener('click', () => {
-        this._handleDeleteCard();
+        const deleteCardPopup = createPopupWithYes('#delete-card-modal', () => {
+          api.deleteCard(this._data._id);
+          this._handleDeleteCard();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.